src/parser/main.js

```
@@ -27,6 +27,18 @@ function double_quote(text) {
     ]);
 }
 
+function mkRedirect(token, nodes, kind) {
+    var m = token.match(/^([0-9]*)(?:>>?|<)$/);
+    var fd = (m && m[1]) ? m[1] : (kind === 'input' ? '0' : '1');
+    return { type: kind, fd: fd, value: nodes };
+}
+
+function mkDup(token) {
+    var m = token.match(/^([0-9]*?)>&([0-9]+)$/);
+    var fd = (m && m[1] && m[1].length) ? m[1] : '1';
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)